Src/FluentAssertions/NumericAssertionsExtensions.cs

```diff
@@ -1481,6 +1481,209 @@ public static class NumericAssertionsExtensions
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public static AndConstraint<NumericAssertions<double>> NotBeNaN(this NumericAssertions<double> parent,
+        string because = "",
+        params object[] becauseArgs)
+    {
+        double actualValue = parent.Subject.Value;
+
+        FailIfValueIsNaN(double.IsNaN(actualValue), because, becauseArgs);
+
+        return new AndConstraint<NumericAssertions<double>>(parent);
+    }
+
+    /// <summary>
+    /// Asserts that the float value is not NaN.
+    /// </summary>
+    /// <param name="parent">The <see cref="NumericAssertions{T}"/> object that is being extended.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public static AndConstraint<NullableNumericAssertions<float>> NotBeNaN(this NullableNumericAssertions<float> parent,
+        string because = "",
+        params object[] becauseArgs)
+    {
+        float? actualValue = parent.Subject;
+
+        FailIfValueIsNaN(actualValue is { } value && float.IsNaN(value), because, becauseArgs);
+
+        return new AndConstraint<NullableNumericAssertions<float>>(parent);
+    }
+
+    /// <summary>
+    /// Asserts that the double value is not NaN.
+    /// </summary>
+    /// <param name="parent">The <see cref="NumericAssertions{T}"/> object that is being extended.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public static AndConstraint<NullableNumericAssertions<double>> NotBeNaN(this NullableNumericAssertions<double> parent,
+        string because = "",
+        params object[] becauseArgs)
+    {
+        double? actualValue = parent.Subject;
+
+        FailIfValueIsNaN(actualValue is { } value && double.IsNaN(value), because, becauseArgs);
+
+        return new AndConstraint<NullableNumericAssertions<double>>(parent);
+    }
+
+    private static void FailIfValueIsNaN(bool valueIsNaN,
```

Comment:
🤔 I'm not a big fan of applying DRY like that. In most cases, duplicating the code makes the implementation much more readable.